dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEventQueryPlanner.java

```diff
@@ -242,7 +244,10 @@ else if ( params.isCollapseDataDimensions() && !params.getItems().isEmpty() )
     {
         List<EventQueryParams> queries = new ArrayList<>();
         
-        if ( params.getAggregationTypeFallback().isLastPeriodAggregationType() && !params.getPeriods().isEmpty() )
+        
+        if ( ( params.getAggregationType() != null && params.getAggregationType().isLastPeriodAggregationType() ) || 
```

Comment:
Add utility methods on DataQueryParams to remove 2 of these conditions, makes it more readable.